### ISSUE TITLE
[0.9.3] wal compacting causing database to time out

### ISSUE THREAD

**Issue by kenzodeluxe** [influxdb/influxdb#4022]
Once a database has reached a critical size (11GB was mentioned in another issue and seems relevant somehow), wal flushing is causing outages:
```
[http] 2015/09/06 23:56:55 1.2.3.4 - - [06/Sep/2015:23:56:55 +0200] POST /write?db=monitoring&precision=s HTTP/1.1 204 0 - client 20.828986ms
[wal] 2015/09/06 23:56:56 Flush due to threshold. Flushing 982 series with 5260667 bytes from partition 1. Compacting 702 series
(..)
[http] 2015/09/06 23:57:01 1.2.3.4 - - [06/Sep/2015:23:56:56 +0200] POST /write?db=monitoring&precision=s HTTP/1.1 500 32 - client 5.007814203s
(...)
[http] 2015/09/07 00:02:34 1.2.3.4 - - [07/Sep/2015:00:02:29 +0200] POST /write?db=monitoring&precision=s HTTP/1.1 500 32 - client 5.008541152s

```
The database will never recover from this fault until it's removed or wal is deleted. Will then happen again once wal is flushed again.
du -hs:
55G	/var/opt/influxdb/data/monitoring

**Comment by kenzodeluxe** [comment]
The following changes have been added to the config:

wal-ready-series-size = 4096
wal-partition-size-threshold = 10485760

**Comment by kenzodeluxe** [comment]
Neither writing, nor reading is possible when this state has been reached. At a certain level (>55GB?) once dropping the database appears to help.

**Comment by beckettsean** [comment]
@kenzodeluxe what version of the database? 0.9.3, nightly build, or master?

**Comment by beckettsean** [comment]
Sorry, I missed the header, nevermind.

**Comment by pauldix** [comment]
After a bunch of testing, we've determined that the underlying index just can't keep up with the insertion load that the WAL makes possible. To prevent the system from going OOM and dying we're going to put back pressure on the write clients and start failing writes when memory gets too high. See #4066.

We're also working on a massive storage engine upgrade that will fix this. Should be available in the nightlies within a week or two. It will have much better compression and be able to keep up with the write load of the WAL.

More soon...

**Comment by kenzodeluxe** [comment]
Thanks! Looking forward to getting the fix.

**Comment by otoolep** [comment]
0.9.4-rc1 contains the change that writes will start failing if memory usage gets too high. 

  https://s3.amazonaws.com/influxdb/influxdb_0.9.4-rc1_amd64.deb
  https://s3.amazonaws.com/influxdb/influxdb-0.9.4_rc1-1.x86_64.rpm

**Comment by kenzodeluxe** [comment]
Another finding: When compacting leads to 500s for _all_ databases, _sometimes_ restarting influxdb does the trick. Hope that this matches your findings and the fix you mentioned is suitable for this situation, too.

**Comment by kenzodeluxe** [comment]
Appears to be fixed in 0.9.4.1.
